Src/main.c

```diff
@@ -484,7 +488,7 @@ const char *noteI2CReceive(uint16_t DevAddress, uint8_t* pBuffer, uint16_t Size,
             errstr = NULL;
             break;
         }
-        errstr = "i2c: write error";
+        errstr = "i2c: read error";
```

Comment:
I'm not saying you're wrong here, but would you expand on your reasoning?

This error message is attached to a call to `HAL_I2C_Master_Transmit()`

Reply:
At the lowest level it is actually a write error, but at a high level it is a read error, so I can both agree and disagree with your error message.

Tell me why you like it better this way.